cmd/samples/http/responder/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client/http"
+	cehttp "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/kelseyhightower/envconfig"
+	"log"
+	"os"
+)
+
+type envConfig struct {
+	// Port on which to listen for cloudevents
+	Port int    `envconfig:"RCV_PORT" default:"8080"`
+	Path string `envconfig:"RCV_PATH" default:"/"`
+}
+
+func main() {
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Printf("[ERROR] Failed to process env var: %s", err)
+		os.Exit(1)
+	}
+	os.Exit(_main(os.Args[1:], env))
+}
+
+type Example struct {
+	Sequence int    `json:"id"`
+	Message  string `json:"message"`
+}
+
+func gotEvent(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
+	fmt.Printf("Got Event Context: %+v\n", event.Context)
+	data := &Example{}
+	if err := event.DataAs(data); err != nil {
+		fmt.Printf("Got Data Error: %s\n", err.Error())
+	}
+	fmt.Printf("Got Data: %+v\n", data)
+	fmt.Printf("Got Transport Context: %+v\n", http.TransportContextFrom(ctx))
+	fmt.Printf("----------------------------\n")
+
+	if data.Sequence%3 == 0 {
+		r := cloudevents.Event{
+			Context: cloudevents.EventContextV02{
+				Source: *types.ParseURLRef("/mod3"),
+				Type:   "samples.http.mod3",
+			}.AsV02(),
+			Data: Example{
+				Sequence: data.Sequence,
+				Message:  "mod 3!",
+			},
+		}
+		resp.RespondWith(200, &r)
+		return nil
+	}
+
+	return nil
+}
+
+func _main(args []string, env envConfig) int {
+	ctx := context.Background()
+
+	c, err := http.New(
+		cehttp.WithPort(env.Port),
+		cehttp.WithPath(env.Path),
+		client.WithUUIDs(),
+		client.WithTimeNow(),
```

Comment:
This is REALLY subtle and awesome. I am setting client options on the http client wrapper, and using those event defaulters to default values on the response event. 💯